### PR TITLE
doc: add developer guidelines

### PR DESCRIPTION
**Pull request by alban**
It's not obvious how the TAP output is displayed and the meaning of the
exit value.

This documentation is meant to be for developers who want to add a new
validation test.

Signed-off-by: Alban Crequy <alban@kinvolk.io>

### CODE REVIEW

File: docs/devel_guidelines.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+### TAP output
+
+Each validation test prints TAP output.
+So far, we have two kinds of validation tests and they print the TAP output differently:
+* tests using `util.RuntimeInsideValidate`: they start the process `runtimetest` inside the container and `runtimetest` prints the TAP output. The test process itself must not output anything to avoid mixing its output with the TAP output. Each test can only call `util.RuntimeInsideValidate` one time because several TAP outputs cannot be concatenated.
+* tests using `util.RuntimeOutsideValidate`: they create a container but without executing `runtimetest`. The test program itself must print the TAP output.
+
+### Exit status
+
+When the runtime fails a test, the TAP output indicates so with "not ok" but the exit status of test program normally remain 0.
```
Comment (wking) [inline docs/devel_guidelines.md]: "remain" -> "remains".

File: docs/devel_guidelines.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+A non-zero exit status indicates a problem in the test program rather than in the runtime.
+
```
Comment (wking) [inline docs/devel_guidelines.md]: No need for this trailing empty line.

**Comment by alban** [comment]
@wking thanks for the review! I updated the commit.

**Comment by q384566678** [comment]
LGTM

**Pull request merged**